config/config.go

```diff
@@ -12,594 +12,91 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-// Package config stores and retrieves configuration settings for the OS Config agent.
 package config
 
 import (
 	"context"
-	"crypto/sha256"
-	"encoding/json"
-	"flag"
+	"errors"
 	"fmt"
-	"io/ioutil"
-	"net"
-	"net/http"
-	"net/url"
-	"os"
-	"runtime"
-	"strconv"
-	"strings"
-	"sync"
-	"time"
 
-	"cloud.google.com/go/compute/metadata"
-	"github.com/GoogleCloudPlatform/osconfig/clog"
-	"golang.org/x/oauth2/jws"
+	agentendpointpb "github.com/GoogleCloudPlatform/osconfig/internal/google.golang.org/genproto/googleapis/cloud/osconfig/agentendpoint/v1alpha1"
 )
 
-const (
-	// metadataIP is the documented metadata server IP address.
-	metadataIP = "169.254.169.254"
-	// metadataHostEnv is the environment variable specifying the
-	// GCE metadata hostname.
-	metadataHostEnv = "GCE_METADATA_HOST"
-	// InstanceMetadata is the compute metadata URL.
-	InstanceMetadata = "http://metadata.google.internal/computeMetadata/v1/instance"
-	// IdentityTokenPath is the instance identity token path.
-	IdentityTokenPath = "instance/service-accounts/default/identity?audience=osconfig.googleapis.com&format=full"
-	// ReportURL is the guest attributes endpoint.
-	ReportURL = InstanceMetadata + "/guest-attributes"
+// Resource is a single config resource.
+type Resource struct {
```

Comment:
Would it be better to qualify this as ConfigResource or OSPolicyResource since we also have another interface name resource

Reply:
renamed to OSPolicyResource as I think thats the naming we want to use